### PR TITLE
Replace shell.nix with a updated flake.nix

### PR DESCRIPTION
**Pull request by LogicalOverflow**
Replaces the `shell.nix` with a flake, that provides a devShell with the libraries and tools need to build pinnacle. It also sets up the `LD_LIBRARY_PATH` so pinnacle can start.

It does not include a nix package for pinnacle (yet), as the build process isn't quite trivial (with it generating configs and doing lua things I don't know anything about). I might tackle that at a later time.

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,59 @@
+{
+  description = " A WIP Smithay-based Wayland compositor, inspired by AwesomeWM and configured in Lua or Rust Topics";
```
Comment (Ottatop) [inline flake.nix]:
```suggestion
  description = " A WIP Smithay-based Wayland compositor, inspired by AwesomeWM and configured in Lua or Rust";
```

Copy paste error, I presume

**Comment by Ottatop** [comment]
Oh also rebase/merge with main and we should be good

File: flake.nix (hunk 1 above)
Comment (LogicalOverflow) [reply flake.nix]: Indeed; fixed

**Comment by LogicalOverflow** [comment]
Just rebased the PR on main

**Review by Ottatop — APPROVED** [review]

**Comment by Ottatop** [comment]
Awesome, thank you!

**Pull request merged**